### ISSUE TITLE
OpenDocumentResult to call OnCompleted immediately, not on deactivation of document

### ISSUE THREAD

**Issue by rumata28** [tgjones/gemini#4]
My problem is that currently I could not execute any action on the opened document.
For example, I open document, than I would like to load document from file and after that I'd like to parse the file content with parser (to make custom highlight).

But action execution just "hangs" on the line "yield return Show.Document(document);" - that is because OpenDocumentResult threats action completion as moment when user closes the tab.

I understand justification of such behavior for showing modal dialogs, where after showing a dialog, if result of the dialog was successful, another operation may need to be run, but operations of opening textual files does not need such behavior.

So, it would be nice to have, at least, an option - if after activating the document result is completed or it shall wait on document deactivation.
